eth/blockwatch/block_watcher.go

```diff
@@ -637,7 +637,12 @@ func (w *Watcher) enrichWithL1BlockNumber(header *MiniHeader) (*MiniHeader, erro
 	if header == nil || header.L1BlockNumber != nil {
 		return header, nil
 	}
-	return w.client.HeaderByHash(header.Hash)
+	fetchedBlock, err := w.client.HeaderByHash(header.Hash)
+	if err != nil {
+		return header, err
+	}
+	header.L1BlockNumber = fetchedBlock.L1BlockNumber
```

Comment:
Ah so the problem was that if `header` contained logs that were previously added to it we could end up returning a MiniHeader returned by HeaderByHash that does not have those logs. And the fix here is to make sure we copy over the L1BlockNumber from the MiniHeader returned by HeaderByHash to the `header` that is passed in to ensure that if the `header` passed in contains logs that they are preserved?

Reply:
Yes, exactly as you described. Before, I thought that what we get from the RPC call already include in logs, but apparently that's not the case.